src/lib.rs

```diff
@@ -17,87 +21,189 @@ impl Default for Syntax {
+    /// your field is likely to have negative values, use sint32 instead.
     Int32,
+    /// Protobuf int64
+    ///
+    /// # Remarks
+    ///
+    /// Uses variable-length encoding. Inefficient for encoding negative numbers – if
+    /// your field is likely to have negative values, use sint64 instead.
     Int64,
+    /// Protobuf uint32
+    ///
+    /// # Remarks
+    ///
+    /// Uses variable-length encoding.
     Uint32,
+    /// Protobuf uint64
+    ///
+    /// # Remarks
+    ///
+    /// Uses variable-length encoding.
     Uint64,
+    /// Protobuf sint32
+    ///
+    /// # Remarks
+    ///
+    /// Uses variable-length encoding. Signed int value. These more efficiently
+    /// encode negative numbers than regular int32s.
     Sint32,
+    /// Protobuf sint64
+    ///
+    /// # Remarks
+    ///
+    /// Uses variable-length encoding. Signed int value. These more efficiently
+    /// encode negative numbers than regular int32s.
     Sint64,
+    /// Protobuf bool
     Bool,
+    /// Protobuf enum (holds the enum name)
     Enum(String),
+    /// Protobuf fixed64
+    ///
+    /// # Remarks
+    ///
+    /// Always eight bytes. More efficient than uint64 if values are often greater than 2^56.
     Fixed64,
+    /// Protobuf sfixed64
+    ///
+    /// # Remarks
+    ///
+    /// Always eight bytes.
     Sfixed64,
+    /// Protobuf double
     Double,
+    /// Protobuf string
+    ///
+    /// # Remarks
+    ///
+    /// A string must always contain UTF-8 encoded or 7-bit ASCII text.
     String_,
+    /// Protobuf bytes
+    ///
+    /// # Remarks
+    ///
+    /// May contain any arbitrary sequence of bytes.
     Bytes,
+    /// Protobut message (holds the message name)
     Message(String),
+    /// Protobut fixed32
+    ///
+    /// # Remarks
+    ///
+    /// Always four bytes. More efficient than uint32 if values are often greater than 2^28.
     Fixed32,
+    /// Protobut sfixed32
+    ///
+    /// # Remarks
+    ///
+    /// Always four bytes.
     Sfixed32,
+    /// Protobut float
     Float,
+    /// Protobut map
     Map(Box<(FieldType, FieldType)>),
 }
 
+/// A Protobuf Field
 #[derive(Debug, Clone)]
 pub struct Field {
+    /// Field name
     pub name: String,
+    /// Field `Frequency`
     pub frequency: Frequency,
+    /// Field type
     pub typ: FieldType,
+    /// Tag number
     pub number: i32,
+    /// Default value for the field
     pub default: Option<String>,
+    /// Packed property for repeated fields
     pub packed: Option<bool>,
-    pub boxed: bool,
+    /// Is the field deprecated
     pub deprecated: bool,
 }
 
+/// A protobuf message
 #[derive(Debug, Clone, Default)]
 pub struct Message {
+    /// Message name
     pub name: String,
+    /// Message `Field`s
     pub fields: Vec<Field>,
+    /// Message `OneOf`s
     pub oneofs: Vec<OneOf>,
+    /// Message reserved numbers
     pub reserved_nums: Option<Vec<i32>>,
```

Comment:
IMHO this can be simply `Vec<i32>`.

Also, in `descriptor.proto` reserved fields are list of structs, not a list of values:

```
  message ReservedRange {
    optional int32 start = 1; // Inclusive.
    optional int32 end = 2;   // Exclusive.
  }
  repeated ReservedRange reserved_range = 9;
```

Reply:
> IMHO this can be simply Vec<i32>, without Option.

Ok. I don't mind.

> Also, in descriptor.proto reserved fields are list of structs, not a list of values:

Ok. I'll use a `Vec<Range<i32>>`.